index.bs

```diff
@@ -1078,38 +1095,150 @@ Issue: Define how this works.
 Each EMCAScript [=Realm=] has a corresponding <dfn>handle object map</dfn>. This is a
 strong map from handle ids to their corresponding objects.
 
+Each {{Window}} with [=parent=] <code>null</doce> has a corresponding
+<dfn>platform objects map</dfn>. This is a weak map from shared ids to their
+corresponding [=platform object=].
+
 <pre class="cddl remote-cddl">
 OwnershipModel = "root" / "none";
 </pre>
 
-<dfn>RemoteReference</dfn> represents a remote reference to an existing ECMAScript
-object in [=handle object map=] in the given [=Realm=].
+<dfn>RemoteReference</dfn> is either a <code>RemoteObjectReference</code>
+representing a remote reference to an existing ECMAScript object in
+[=handle object map=] in the given [=Realm=], or is a <code>SharedReference</code>
+representing a reference to a [=platform object=] in the given
+[=/browsing context=]'s [=list of known elements=], [=list of known shadow roots=] or
+in the top level {{Window}}'s [=platform objects map=].
 
 <pre class="cddl remote-cddl local-cddl">
+RemoteReference = {
+  RemoteObjectReference //
+  SharedReference
+}
+
 Handle = text;
 
-RemoteReference = {
+RemoteObjectReference = {
    handle: Handle,
    Extensible
 }
+
+SharedId = text;
+
+SharedReference = {
+   sharedId: SharedId,
+   Extensible
+}
 </pre>
 
 Issue: handle "stale object reference" case.
 
+Note: if the provided reference has both <code>handle</code> and
```

Comment:
This is a case where I especially don't love this strategy of allowing ambiguous input and just picking a processing order to resolve the ambiguity. In this case I'd almost prefer that we defined things like
```
RemoteObjectReference = {
   handle: Handle,
   ?sharedId: SharedId
   Extensible
}

SharedReference = {
   ?handle: Handle,
   sharedId: SharedId
   Extensible
}
```
and then if you get both fields check that they actually refer to the same object.

Reply:
> This is a case where I especially don't love this strategy of allowing ambiguous input and just picking a processing order to resolve the ambiguity. In this case I'd almost prefer that we defined things like.

Changed the signature.

> if you get both fields check that they actually refer to the same object

This approach would not allow re-using the serialized node in another sandbox, as the handle would not point to any value.

Reply:
> This approach would not allow re-using the serialized node in another sandbox, as the handle would not point to any value.

It would mean that you couldn't provide an unusable handle along with the sharedId, in case you were trying to use things in a different sandbox. So yes, in that case you'd have to remove the field. But this is why I don't really like the DWIM approach to API design where we allow ambiguous protocol messages and hope that we define the right resolution rules to work how people expect most of the time, vs just having an unambigous protocol but requiring some extra upfront work to make the payloads match the format.

Reply:
We can do the following:
1. Move this question to another issue.
1. Ask for client's feedback.
1. Change the spec if required to something like

```
RemoteObjectReference = {
   type: "handle",
   handle: Handle,
}

SharedReference = {
   type: "sharedReference",
   sharedId: SharedId
}
```

WDYT?